website/reorg-add-concepts-tasks-tutorials-reference.md

```diff
@@ -0,0 +1,184 @@
+# Reorganize to use K8s-inspired concepts/tasks/tutorials/reference sections
+
+## Summary
+
+With our current Getting Started and Operations sections, it is increasingly difficult to scale to add new material without dealing with
+significant duplication and ill-fitting organization. Follow the experience of the main Kubernetes documentation by organizing this content
+in Concepts, Tasks, Tutorials, and Reference sections.
+
+## Proposal
+
+Reorganize `submariner-website/src/content` to:
+
+```text
+.
+├── _index.md
+├── getting-started
+│   ├── _index.md
+│   └── known-issues.md # TODO: Combine into index page?
+├── concepts
+│   ├── _index.md
+│   ├── architecture
+│   │   ├── _index.md
+│   │   ├── broker.md
+│   │   ├── gateway-engine.md
+│   │   ├── globalnet.md
+│   │   ├── route-agent.md
+│   │   └── service-discovery.md
+│   ├── clusterset.md
+│   └── serviceexport-serviceimport.md
+├── tasks
+│   ├── _index.md
+│   ├── infra
+│   │   ├── _index.md
+│   │   ├── aws.md
+│   │   ├── calico.md
+│   │   ├── gke.md
+│   │   ├── kind.md
+│   │   ├── openshift.md
+│   │   └── rancher.md
+│   ├── deploy
+│   │   ├── _index.md
+│   │   ├── helm
+│   │   │   ├── _index.md
+│   │   │   ├── install-charts.md
+│   │   │   ├── broker.md
+│   │   │   └── join-clusterset.md
+│   │   └── subctl
+│   │       ├── _index.md
+│   │       ├── install-subctl.md
+│   │       ├── broker.md
+│   │       ├── join-clusterset.md
+│   │       └── submariner-with-sd.md
+│   ├── verify
+│   │   ├── _index.md
+│   │   ├── subctl-verify.md
+│   │   ├── manual-service-connectivity.md
+│   │   └── manual-statefulset.md
+│   ├── monitor
+│   │   ├── _index.md
+│   │   └── enable-prometheus.md
+│   └── troubleshoot
+│       ├── _index.md
+│       ├── connectivity.md
+│       └── service-discovery.md
+├── tutorials
+│   ├── _index.md
+│   └── # TODO Nir's WIP "user guide" PR reformatted
+├── reference
+│   ├── _index.md
+│   ├── prometheus-metrics.md
+│   └── subctl.md
+├── community
+│   ├── _index.md
+│   ├── code-of-conduct.md
+│   ├── contributor-roles.md
+│   ├── getting-help.md
+│   ├── releases.md
+│   └── roadmap.md
+├── development
+│   ├── _index.md
+│   ├── building-testing.md
+│   ├── code-review.md
+│   ├── release-process.md
+│   ├── security.md
+│   ├── shipyard
+│   │   ├── _index.md
+│   │   ├── first-time.md
+│   │   └── advanced.md
+│   └── website
+│       ├── _index.md
+│       └── style_guide.md
```

Comment:
Following up the discussion on https://github.com/submariner-io/submariner-website/pull/708 I think it's better to split this section (or nest it) since there are two types of documentation here:
1. General guidelines (security, code review, contribution guide (which isn't listed here), website style guide).
2. Specific operational guides (building and testing, releasing, everything shipyard, website contributions)

We could create either two sections on the top level, or nest on inside the other.

If we're following K8s docs example, the first category would be akin to "Contribute" and for the second one I dont see on a formal website but is akin to https://github.com/kubernetes/community/tree/master/contributors/devel

For example, we could have a structure such as:
```suggestion
├── contributing
│   ├── _index.md
│   ├── concepts
│   │   ├── code-review.md
│   │   ├── contributing.md # https://submariner.io/development/contribution-guide/
│   │   └── security.md
│   └── tasks
│       ├── building-testing.md
│       ├── release-process.md
│       ├── shipyard
│       │   ├── _index.md
│       │   ├── first-time.md
│       │   └── advanced.md
│       └── website
│           ├── _index.md
│           └── style_guide.md
```